runtime/stdlib/test.go

```diff
@@ -0,0 +1,472 @@
+		blockchainConstructorVar := testContract.NestedVariables[blockchainTypeName]
+		blockchainConstructor, ok := blockchainConstructorVar.GetValue().(*interpreter.HostFunctionValue)
+		if !ok {
+			panic(errors.NewUnexpectedError("invalid type for constructor"))
+		}
+
+		blockchain, err := inter.InvokeExternally(
+			blockchainConstructor,
+			blockchainConstructor.Type,
+			[]interpreter.Value{
+				emulatorBackend,
+			},
+		)
+
+		if err != nil {
+			panic(err)
+		}
+
+		return blockchain
+	},
+	testNewEmulatorBlockchainFunctionType,
+)
+
+// 'EmulatorBackend' struct.
+//
+// 'EmulatorBackend' is the native implementation of the 'Test.BlockchainBackend' interface.
+// It provides a blockchain backed by the emulator.
+
+const emulatorBackendTypeName = "EmulatorBackend"
+
+var EmulatorBackendType = func() *sema.CompositeType {
+
+	ty := &sema.CompositeType{
+		Identifier: emulatorBackendTypeName,
+		Kind:       common.CompositeKindStructure,
+		Location:   TestContractLocation,
+		ExplicitInterfaceConformances: []*sema.InterfaceType{
+			blockchainBackendInterfaceType,
+		},
+	}
+
+	var members = []*sema.Member{
+		sema.NewUnmeteredPublicFunctionMember(
+			ty,
+			emulatorBackendExecuteScriptFunctionName,
+			emulatorBackendExecuteScriptFunctionType,
+			emulatorBackendExecuteScriptFunctionDocString,
+		),
+	}
+
+	ty.Members = sema.GetMembersAsMap(members)
+	ty.Fields = sema.GetFieldNames(members)
+
+	return ty
+}()
+
+func newEmulatorBackend(
+	inter *interpreter.Interpreter,
+	getLocationRange func() interpreter.LocationRange,
+) *interpreter.CompositeValue {
+
+	var fields = []interpreter.CompositeField{
+		{
+			Name:  emulatorBackendExecuteScriptFunctionName,
+			Value: emulatorBackendExecuteScriptFunction,
+		},
+	}
+
+	return interpreter.NewCompositeValue(
+		inter,
+		getLocationRange,
+		EmulatorBackendType.Location,
+		emulatorBackendTypeName,
+		common.CompositeKindStructure,
+		fields,
+		common.Address{},
+	)
+}
+
+// 'EmulatorBackend.executeScript' function
+
+const emulatorBackendExecuteScriptFunctionName = "executeScript"
+
+const emulatorBackendExecuteScriptFunctionDocString = `execute script function`
+
+var emulatorBackendExecuteScriptFunctionType = func() *sema.FunctionType {
+	// The type of the 'executeScript' function of 'EmulatorBackend' (interface-implementation)
+	// is same as that of 'BlockchainBackend' interface.
+	typ, ok := blockchainBackendInterfaceType.Members.Get(emulatorBackendExecuteScriptFunctionName)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"cannot find type %s.%s",
+			blockchainBackendTypeName,
+			emulatorBackendExecuteScriptFunctionName,
+		))
+	}
+
+	functionType, ok := typ.TypeAnnotation.Type.(*sema.FunctionType)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"invalid type for %s. expected function",
+			emulatorBackendExecuteScriptFunctionName,
+		))
+	}
+
+	return functionType
+}()
+
+var emulatorBackendExecuteScriptFunction = interpreter.NewUnmeteredHostFunctionValue(
```

Comment:
Instead of making `Interpreter` know about a testing framework by adding a `TestFramework` field to it, just to be able to get access in this host function, maybe (if possible) make `emulatorBackendExecuteScriptFunction` a function that gets a test framework and returns a host function (`func newEmulatorBackendExecuteScriptFunction(testFramework TestFramework) *interpreter.HostFunctionValue`)

Reply:
Yes, should be possible.

Unfortunately the follow-up PRs also depend on this pattern and there is some refactoring around the test runner which could severely conflict with this change. I will refactor once the first four PRs (up to adding transaction support) are merged. That way I can also change it once for all test-stdlib functions. WDYT?

Reply:
Of course, please feel free to do work in a follow-up PR if that makes life easier 👍